src/frontend/telemetry.rs

```diff
@@ -0,0 +1,146 @@
+use clarity_repl::clarity::util::hash::{bytes_to_hex, Hash160};
+use mac_address::get_mac_address;
+use segment::{
+    message::{Message, Track, User},
+    Client, HttpClient,
+};
+
+use crate::publish::Network;
+
+pub enum DeveloperUsageEvent {
+    NewProject(DeveloperUsageDigest),
+    PokeExecuted(DeveloperUsageDigest),
+    CheckExecuted(DeveloperUsageDigest),
+    TestSuiteExecuted(DeveloperUsageDigest, bool, u32),
+    DevnetExecuted(DeveloperUsageDigest),
+    ContractPublished(DeveloperUsageDigest, Network),
+    UnknownCommand(DeveloperUsageDigest, String),
+}
+
+pub struct DeveloperUsageDigest {
+    pub project_id: String,
+    pub team_id: String,
+}
+
+impl DeveloperUsageDigest {
+    pub fn new(project_id: &str, team_id: &Vec<String>) -> Self {
+        let hashed_project_id = Hash160::from_data(project_id.as_bytes());
+        let hashed_team_id = Hash160::from_data(team_id.join(",").as_bytes());
+        Self {
+            project_id: format!("0x{}", bytes_to_hex(&hashed_project_id.to_bytes().to_vec())),
+            team_id: format!("0x{}", bytes_to_hex(&hashed_team_id.to_bytes().to_vec())),
+        }
+    }
+}
+
+pub fn telemetry_report_event(event: DeveloperUsageEvent) {
+    let rt = tokio::runtime::Builder::new_current_thread()
+        .enable_io()
+        .enable_time()
+        .max_blocking_threads(32)
+        .build()
+        .unwrap();
+
+    rt.block_on(send_event(event));
+}
+
+async fn send_event(event: DeveloperUsageEvent) {
+    let segment_api_key = "<REPLACE_SEGMENT_API_KEY>";
```

Comment:
Is this secure? I would expect we should only have an API key embedded if this was running server-side for security reasons.

Reply:
Oh, I see... is this just the write key, which is intended to be public?

Reply:
This is a placeholder that the CI should be seeding (via a Github Secret) before building releases. I might have to summon @CharlieC3 for the last mile on this one.

Reply:
Right, I figured that the CI job would be filling it in. I was just worried about putting an API key into the deployed binary, since if it is in there, it could be extracted. After I looked at the Segment docs I realized that it looks like the "write key" is expected to be public, so that should not be a problem. 

Actually, since that is the case, I guess it should be okay to just include it in the repo, right?

Reply:
@obycode Can you point out where it says the API key is expected to be public? My understanding is that anyone that has the API key can write to our Segment account.

Reply:
They have to be public because they can be used on a frontend to send telemetry to Segment, so there is no choice but to expose them in the JS code. 

Here's something from an incident report:
> ... including customer write keys for Segment (which are considered public),...
https://segment.com/blog/bulletins-incident090519/

I don't see it in the main docs, but it makes sense if they're telling you to use them on the client side, they have to be exposed.

Reply:
Yes, anyone with that write key could send fake data to our Segment account. The hope is just that there is no motivation to do so. I guess it doesn't hurt if we leave it out of the repo, so an attacker would at least have to do a little work to extract the key from the binary.

Reply:
The only problem will be if someone builds from source and enables telemetry, they'll have to fill in that field or it will get errors all the time.

Reply:
Interesting, thanks for the info! There may still be some value in not making it easy for anyone to obtain (e.g. plaintext in the repo), so it could still be injected during CI. But we'd need to be sure people can still build clarinet locally without needing to supply  a Segment API key. Is that something we can ensure?

Reply:
Looks like we had the same thought there 😄. We could just add a note in the README warning about enabling telemetry when building from source.

Reply:
ha you beat me to it by seconds! 😆 
Yeah that sounds good. Is telemetry enabled by default? Maybe it should be disabled by default so users don't `git pull` the latest changes and have to do something to mitigate them.